### PR TITLE
fix(publish): issue where publish was always true

### PR DESCRIPTION
**Pull request by dmccaffery**
* resolve `IsPublishable` being set to true by SDK targets in new parser
* pin all versions
* make dependency versions configurable
* make `AM.Condo.Test` packable with new xunit version

### CODE REVIEW

File: src/AM.Condo.Abstractions/AM.Condo.Abstractions.csproj [hunk 1]
```diff
@@ -5,9 +5,12 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="System.Runtime.Extensions" Version="[4.3.0, 5.0.0)" />
-    <PackageReference Include="System.Collections" Version="[4.3.0, 5.0.0)" />
-    <PackageReference Include="NuGet.Versioning" Version="[4.4.0, 5.0.0)" />
+    <PackageReference Include="System.Runtime.Extensions" Version="4.3.0" />
```
Comment (sjk07) [inline src/AM.Condo.Abstractions/AM.Condo.Abstractions.csproj:8]: @dmccaffery any reason why we are version locking now?

Comment (dmccaffery) [reply src/AM.Condo.Abstractions/AM.Condo.Abstractions.csproj:8]: @sjk07 it doesn't actually version lock anymore; binding redirects were reintroduced and are autogenerated. scoping ranges doesn't have much effect anymore.

Comment (dmccaffery) [reply src/AM.Condo.Abstractions/AM.Condo.Abstractions.csproj:8]: The other reason is: we control the environment in which condo runs, now -- within docker; no reason not to pin the versions if we can -- avoids nuance bugs (although binding redirects will happen anyway)

Comment (sjk07) [reply src/AM.Condo.Abstractions/AM.Condo.Abstractions.csproj:8]: sounds magical.

**Review by sjk07 — APPROVED** [review]

**Pull request merged**